Index: u_main.tk
===================================================================
RCS file: /cvsroot/pure-data/pd/src/u_main.tk,v
retrieving revision 1.25
diff -u -w -r1.25 u_main.tk
--- u_main.tk	9 Oct 2006 04:36:12 -0000	1.25
+++ u_main.tk	10 Oct 2007 05:22:28 -0000
@@ -549,6 +549,18 @@
         -command {menu_doc_open doc/1.manual index.htm} 
     $mbar.help add command -label {Browser ...} \
         -command {menu_doc_browser $help_top_directory} 
+    $mbar.help add separator
+    $mbar.help add command -label {puredata.org} \
+        -command {menu_openhtml http://puredata.org} 
+    $mbar.help add command -label {Pdpedia} \
+        -command {menu_openhtml http://pdpedia.org} 
+    $mbar.help add separator
+    $mbar.help add command -label {mailing lists} \
+        -command {menu_openhtml http://puredata.org/community/lists} 
+    $mbar.help add command -label {forums} \
+        -command {menu_openhtml http://puredata.org/community/forums} 
+    $mbar.help add command -label {FAQ} \
+        -command {menu_openhtml http://puredata.org/docs/faq} 
 }
 
 #################### the "File" menu for the Pd window ##############
