Index: u_main.tk
===================================================================
--- u_main.tk	(revision 9735)
+++ u_main.tk	(working copy)
@@ -413,7 +413,8 @@
 proc menu_new {} {
     global untitled_number
     global untitled_directory
-    pd [concat pd filename Untitled-$untitled_number $untitled_directory \;]
+    pd [concat pd filename Untitled-$untitled_number \
+			[pdtk_enquote $untitled_directory] \;]
     pd {
         #N canvas;
         #X pop 1;
