Index: s_main.c
===================================================================
RCS file: /cvsroot/pure-data/pd/src/s_main.c,v
retrieving revision 1.22
diff -u -w -r1.22 s_main.c
--- s_main.c	31 Dec 2005 20:55:25 -0000	1.22
+++ s_main.c	27 Feb 2006 17:04:49 -0000
@@ -33,7 +33,9 @@
 int sys_rcfile(void);
 int m_scheduler(void);
 void sys_addhelppath(char *p);
+#ifdef USEAPI_ALSA
 void alsa_adddev(char *name);
+#endif
 
 int sys_debuglevel;
 int sys_verbose;
