Index: s_main.c
===================================================================
RCS file: /cvsroot/pure-data/pd/src/s_main.c,v
retrieving revision 1.28.6.6
diff -u -r1.28.6.6 s_main.c
--- s_main.c	6 Jan 2008 04:19:15 -0000	1.28.6.6
+++ s_main.c	6 Jan 2008 04:29:01 -0000
@@ -249,10 +249,11 @@
 static void pd_makeversion(void)
 {
     char foo[100];
-    sprintf(foo,  "Pd version %d.%d-%d%s\n",PD_MAJOR_VERSION,
-        PD_MINOR_VERSION,PD_BUGFIX_VERSION,PD_TEST_VERSION);
-    pd_version = malloc(strlen(foo)+1);
-    strcpy(pd_version, foo);
+
+    snprintf(foo, sizeof(foo), "Pd version %d.%d-%d%s\n", PD_MAJOR_VERSION,
+        PD_MINOR_VERSION, PD_BUGFIX_VERSION, PD_TEST_VERSION);
+	
+    pd_version = strdup(foo);
 }
 
 /* this is called from main() in s_entry.c */
