Index: s_main.c
===================================================================
RCS file: /cvsroot/pure-data/pd/src/s_main.c,v
retrieving revision 1.20
diff -u -w -r1.20 s_main.c
--- s_main.c	2 Dec 2005 04:58:24 -0000	1.20
+++ s_main.c	22 Feb 2006 05:54:28 -0000
@@ -2,10 +2,6 @@
 * For information on usage and redistribution, and for a DISCLAIMER OF ALL
 * WARRANTIES, see the file, "LICENSE.txt," in this distribution.  */
 
-char pd_version[] = "Pd version 0.39-2\n";
-char pd_compiletime[] = __TIME__;
-char pd_compiledate[] = __DATE__;
-
 #include "m_pd.h"
 #include "m_imp.h"
 #include "s_stuff.h"
@@ -26,6 +22,10 @@
 #include <winbase.h>
 #endif
 
+char *pd_version;
+char pd_compiletime[] = __TIME__;
+char pd_compiledate[] = __DATE__;
+
 void pd_init(void);
 int sys_argparse(int argc, char **argv);
 void sys_findprogdir(char *progname);
@@ -254,6 +254,9 @@
     if (sys_argparse(argc-1, argv+1))           /* parse cmd line */
         return (1);
     sys_afterargparse();                    /* post-argparse settings */
+	/* build version string from defines in m_pd.h */
+	asprintf(&pd_version, "Pd version %d.%d-%d%s\n",PD_MAJOR_VERSION,
+			PD_MINOR_VERSION,PD_BUGFIX_VERSION,PD_TEST_VERSION); 
     if (sys_verbose || sys_version) fprintf(stderr, "%scompiled %s %s\n",
         pd_version, pd_compiletime, pd_compiledate);
     if (sys_version)    /* if we were just asked our version, exit here. */
Index: m_pd.h
===================================================================
RCS file: /cvsroot/pure-data/pd/src/m_pd.h,v
retrieving revision 1.8
diff -u -w -r1.8 m_pd.h
--- m_pd.h	19 Aug 2005 23:28:03 -0000	1.8
+++ m_pd.h	22 Feb 2006 05:54:28 -0000
@@ -10,6 +10,8 @@
 
 #define PD_MAJOR_VERSION 0
 #define PD_MINOR_VERSION 39   
+#define PD_BUGFIX_VERSION 2
+#define PD_TEST_VERSION ""
 
 /* old name for "MSW" flag -- we have to take it for the sake of many old
 "nmakefiles" for externs, which will define NT and not MSW */
