Index: u_main.tk
===================================================================
RCS file: /cvsroot/pure-data/pd/src/u_main.tk,v
retrieving revision 1.25
diff -u -w -r1.25 u_main.tk
--- u_main.tk	9 Oct 2006 04:36:12 -0000	1.25
+++ u_main.tk	15 Aug 2007 22:26:47 -0000
@@ -55,6 +55,11 @@
     set pd_gui2 [string range $argv0 0 [expr [string last / $argv0 ] - 1]]
     set pd_guidir $pd_gui2/..
     load $pd_guidir/bin/libPdTcl.dylib
+# this is the embedded libquicktime can find its embedded plugins
+	 set libquicktime_plugin_dir \
+		  [file normalize $pd_gui2/../../lib/libquicktime]
+	 puts stderr "Setting LIBQUICKTIME_PLUGIN_DIR: $libquicktime_plugin_dir"
+	 set env(LIBQUICKTIME_PLUGIN_DIR) $libquicktime_plugin_dir
     set pd_tearoff 0
     global pd_macready
     set pd_macready 0
