Index: configure.in
===================================================================
RCS file: /cvsroot/pure-data/pd/src/configure.in,v
retrieving revision 1.15
diff -u -w -r1.15 configure.in
--- configure.in	16 Aug 2005 04:06:28 -0000	1.15
+++ configure.in	16 Sep 2006 17:19:16 -0000
@@ -265,7 +265,13 @@
 	-I../portmidi/pm_common -I../portmidi/pm_mac \
 	-I../portmidi/porttime \
 	-Wno-error \
-	-DUSEAPI_PORTAUDIO -DPA_BIG_ENDIAN -DPA19 -DPA_USE_COREAUDIO"
+	-DUSEAPI_PORTAUDIO -DPA19 -DPA_USE_COREAUDIO"
+	 if test `uname -p` == powerpc;
+	 then
+		  MORECFLAGS="$MORECFLAGS -DPA_BIG_ENDIAN"
+	 else
+		  MORECFLAGS="$MORECFLAGS -DPA_LITTLE_ENDIAN"
+	 fi
     SYSSRC="s_midi_pm.c s_audio_pa.c \
     	s_audio_pablio.c     \
 	s_audio_paring.c     \
