*** u_main.tk	2007-10-25 06:18:43.000000000 -0500
--- u_main.tk	2008-04-09 16:49:00.000000000 -0500
***************
*** 39,44 ****
--- 39,56 ----
  
  # end Pd-extended font hacks -----------------------------
  
+ # begin hack to hide hidden files/folder in tk browser on unix ----------
+ if {$pd_nt == 0} {
+     # load the dialog once, otherwise setting the vars will not work
+     catch {tk_getOpenFile -with-invalid-argument} 
+ 
+     # change the environment variables
+     namespace eval ::tk::dialog::file {
+     variable showHiddenBtn 1
+     variable showHiddenVar 0
+     }
+ }
+ # end hidden files/folder hack ------------------
  
  # Tearoff is set to true by default:
  set pd_tearoff 1
