Index: s_path.c
===================================================================
--- s_path.c	(revision 9700)
+++ s_path.c	(working copy)
@@ -380,7 +380,7 @@
     int rcargc;
     char* rcargv[NUMARGS];
     char* buffer;
-    char  fname[MAXPDSTRING], buf[1000], *home = getenv("HOME");
+    char  fname[MAXPDSTRING], buf[1001], *home = getenv("HOME");
     int retval = 1; /* that's what we will return at the end; for now, let's think it'll be an error */
  
     /* initialize rc-arg-array so we can safely clean up at the end */
@@ -424,7 +424,7 @@
     fclose(file);
     if (sys_verbose)
     {
-        if (rcargv)
+        if (rcargc)
         {
             post("startup args from RC file:");
             for (i = 1; i < rcargc; i++)
