Index: d_ugen.c
===================================================================
--- d_ugen.c	(revision 13170)
+++ d_ugen.c	(working copy)
@@ -219,6 +219,11 @@
 
 static void block_bang(t_block *x)
 {
+    if (!dsp_chain) 
+    {
+        pd_error(x, "bang to block~ or switch~ with DSP off has no effect");
+        return;
+    }
     if (x->x_switched && !x->x_switchon)
     {
         t_int *ip;
