Index: g_readwrite.c
===================================================================
--- g_readwrite.c	(revision 9740)
+++ g_readwrite.c	(working copy)
@@ -705,7 +705,11 @@
     {
             /* if not an abstraction, reset title bar and directory */ 
         if (!x->gl_owner)
+        {
             canvas_rename(x, filename, dir);
+            /* update window list in case Save As changed the window name */
+            canvas_updatewindowlist(); 
+        }
         post("saved to: %s/%s", dir->s_name, filename->s_name);
         canvas_dirty(x, 0);
         canvas_reload(filename, dir, &x->gl_gobj);
