Index: s_file.c
===================================================================
RCS file: /cvsroot/pure-data/pd/src/s_file.c,v
retrieving revision 1.9
diff -u -w -r1.9 s_file.c
--- s_file.c	31 Dec 2005 20:55:25 -0000	1.9
+++ s_file.c	23 Jan 2006 02:10:40 -0000
@@ -193,7 +193,7 @@
         post("unable to create registry entry: %s\n", key);
         return;
     }
-    err = RegSetValueEx(hkey, key, 0, REG_SZ, value, strlen(value)+1);
+    err = RegSetValueEx(hkey, key, 0, REG_EXPAND_SZ, value, strlen(value)+1);
     if (err != ERROR_SUCCESS)
         post("unable to set registry entry: %s\n", key);
     RegCloseKey(hkey);
