--- ../../pd/src/u_main.tk	Thu Jun  9 11:54:59 2005
+++ build/Pd.app/Contents/Resources/bin/pd.tk	Thu Jun  9 16:17:58 2005
@@ -37,6 +37,12 @@
 if {$pd_nt == 2} {
 # turn on James Tittle II's fast drawing (wait until I can test this...):
 #   set tk::mac::useCGDrawing 1
+    # set minimun line size for anti-aliasing.  If set to 1 or 0, then every
+    # line will be anti-aliased.  While this makes connections and circles in
+    # [bng] and such look really good, it makes boxes and messages look out of
+    # focus.  Setting this to 2 makes it so the thick audio rate connections
+    # are anti-aliased.  <hans@at.or.at> 2005-06-09
+    set tk::mac::CGAntialiasLimit 2
     global pd_guidir
     global pd_tearoff
     set pd_gui2 [string range $argv0 0 [expr [string last / $argv0 ] - 1]]
@@ -158,7 +164,7 @@
 
 
 frame .printout
-text .printout.text -relief raised -bd 2 -font -*-courier-bold--normal--12-* \
+text .printout.text -relief raised -bd 2 -font { courier 12 } \
     -yscrollcommand ".printout.scroll set" -width 80
 # .printout.text insert end "\n\n\n\n\n\n\n\n\n\n"
 scrollbar .printout.scroll -command ".printout.text yview"
@@ -337,7 +343,7 @@
     global pd_myversion
     set name [format ".help%d" $doc_number]
     toplevel $name
-    text $name.text -relief raised -bd 2 -font -*-courier-bold--normal--12-* \
+    text $name.text -relief raised -bd 2 -font { helvetica 14 } \
     	-yscrollcommand "$name.scroll set" -background white
     scrollbar $name.scroll -command "$name.text yview"
     pack $name.scroll -side right -fill y
